spark-connector/src/test/java/com/datastrato/gravitino/spark/connector/TestSparkTransformConverter.java

```diff
@@ -0,0 +1,229 @@
+        createSortOrders(sortColumnNames, ConnectorConstants.SPARK_DEFAULT_SORT_DIRECTION);
+
+    org.apache.spark.sql.connector.expressions.Transform[] transforms =
+        SparkTransformConverter.toSparkTransform(null, distribution, sortOrders);
+    Assertions.assertTrue(transforms.length == 1);
+    Assertions.assertTrue(transforms[0] instanceof SortedBucketTransform);
+
+    SortedBucketTransform sortedBucketTransform = (SortedBucketTransform) transforms[0];
+    Assertions.assertEquals(bucketNum, (Integer) sortedBucketTransform.numBuckets().value());
+    String[][] sparkSortColumns =
+        JavaConverters.seqAsJavaList(sortedBucketTransform.sortedColumns()).stream()
+            .map(sparkNamedReference -> sparkNamedReference.fieldNames())
+            .toArray(String[][]::new);
+
+    String[][] sparkBucketColumns =
+        JavaConverters.seqAsJavaList(sortedBucketTransform.columns()).stream()
+            .map(sparkNamedReference -> sparkNamedReference.fieldNames())
+            .toArray(String[][]::new);
+
+    Assertions.assertArrayEquals(bucketColumnNames, sparkBucketColumns);
+    Assertions.assertArrayEquals(sortColumnNames, sparkSortColumns);
+  }
+
+  private org.apache.spark.sql.connector.expressions.NamedReference[] createSparkFieldReference(
+      String[][] fields) {
+    return Arrays.stream(fields)
+        .map(field -> FieldReference.apply(String.join(ConnectorConstants.DOT, field)))
+        .toArray(org.apache.spark.sql.connector.expressions.NamedReference[]::new);
+  }
+
+  // split column name for Gravitino
+  private String[][] createGravitinoFieldReferenceNames(String... columnNames) {
+    return Arrays.stream(columnNames)
+        .map(columnName -> columnName.split("\\."))
+        .toArray(String[][]::new);
+  }
+
+  private SortOrder[] createSortOrders(String[][] columnNames, SortDirection direction) {
+    return Arrays.stream(columnNames)
+        .map(columnName -> SortOrders.of(NamedReference.field(columnName), direction))
+        .toArray(SortOrder[]::new);
+  }
+
+  private Distribution createHashDistribution(int bucketNum, String[][] columnNames) {
+    NamedReference[] namedReferences =
+        Arrays.stream(columnNames)
+            .map(columnName -> NamedReference.field(columnName))
+            .toArray(NamedReference[]::new);
+    return Distributions.hash(bucketNum, namedReferences);
+  }
+
+  private void initSparkToGravitinoTransformMap() {
+    sparkToGravitinoPartitionTransformMaps.put(
+        SparkTransformConverter.createSparkIdentityTransform("a"), Transforms.identity("a"));
+    sparkToGravitinoPartitionTransformMaps.put(
+        SparkTransformConverter.createSparkIdentityTransform("a.b"),
+        Transforms.identity(new String[] {"a", "b"}));
+  }
+
+  private void initGravitinoToSparkTransformMap() {
+    gravitinoToSparkPartitionTransformMaps.put(
+        IdentityPartitioningDTO.of("a"), SparkTransformConverter.createSparkIdentityTransform("a"));
```

Comment:
Is there any other way to create a `IdentityPartition` besides using `IdentityPartitioningDTO` @mchades

Reply:
using `com.datastrato.gravitino.rel.expressions.transforms.Transforms#identity(java.lang.String[])`

Reply:
`Transforms#identity` is used when creating tables, here is the test that the transform are generated from load table.

Reply:
It is recommended to use method `Transforms#identity()` first, then convert it to `DTO` using method `toDTO()`, instead of directly using `xxDTO.of`.

Reply:
thx, done